code/modules/dwarven/dwarven_structures.dm

```diff
@@ -0,0 +1,428 @@
+
+/obj/structure/destructible/dwarven/mythril_press/Initialize()
+	AddComponent(/datum/component/material_container,list(/datum/material/gold,/datum/material/silver,/datum/material/titanium,/datum/material/dwarven,), 20000, TRUE, /obj/item/stack/sheet/mineral)
+	. = ..()
+
+/obj/structure/destructible/dwarven/mythril_press/attack_hand(mob/user)
+	var/mob/living/carbon/human/H = user
+	if(!H.has_language(/datum/language/dwarven))
+		to_chat(H, "<span class='notice'>You don't understand how to operate this ancient machinery!</span>")
+		return
+	press_mythril(user)
+	. = ..()
+
+/obj/structure/destructible/dwarven/mythril_press/proc/press_mythril(mob/user)
+	var/datum/component/material_container/materials = GetComponent(/datum/component/material_container)
+	if(materials.has_materials(list(/datum/material/gold= 1000, /datum/material/silver = 1000, /datum/material/titanium = 1000)))
+		materials.use_materials(list(/datum/material/gold= 1000, /datum/material/silver = 1000, /datum/material/titanium = 1000))
+		materials.insert_amount_mat(materials.sheet2amount(1),/datum/material/dwarven)
+		materials.retrieve_all(get_turf(src))
+		to_chat(user, "<span class='notice'>You hear a loud crank as materials are compressed into dwarven alloy!</span>")
+		return
+	materials.retrieve_all(get_turf(src))
+	to_chat(user, "<span class='notice'>The machine makes a loud crank sound, but no alloy falls out!</span>")
+
+/obj/structure/destructible/dwarven/workshop
+	name = "Dwarven workshop"
+	icon_state = "workshop"
+	var/state = 0
+	custom_materials = list(/datum/material/wood = 20000)
+	var/static/list/allowed_types = list(
+		/datum/material/iron,
+		/datum/material/glass,
+		/datum/material/gold,
+		/datum/material/silver,
+		/datum/material/diamond,
+		/datum/material/uranium,
+		/datum/material/plasma,
+		/datum/material/bluespace,
+		/datum/material/bananium,
+		/datum/material/titanium,
+		/datum/material/runite,
+		/datum/material/plastic,
+		/datum/material/adamantine,
+		/datum/material/mythril,
+		/datum/material/wood,
+		/datum/material/dwarven,
+		)
+	var/obj/item/loaded_item
+
+/obj/structure/destructible/dwarven/workshop/Initialize()
+	. = ..()
+	AddComponent(/datum/component/material_container,allowed_types, 20000, TRUE, /obj/item/stack)
+
+/obj/structure/destructible/dwarven/workshop/examine(mob/user)
+	. = ..()
+	. += "<span class='notice'>Alt click to change the function</span>"
+	. += "<span class='notice'>Click with mats to put them inside of the workshop</span>"
+	. += "<span class='notice'>Click with a hand to retrieve all mats</span>"
+	. += "<span class='notice'>Click with a mallet to activate the function</span>"
+	switch(state)
+		if(0)
+			. += "<span class='notice'>Currently it can produce upgrade kits. </span>"
+		if(1)
+			. += "<span class='notice'>Currently it can produce structure prints for creation of new structures. </span>"
+		if(2)
+			. += "<span class='notice'>Current it can produce quality pickaxes </span>"
```

Comment:
bad spaces in all three of these states, no periods in any of the above descriptions